lib/matplotlib/axes/_base.py

```diff
@@ -1191,14 +1191,25 @@ def set_color_cycle(self, clist):
         else:
             self.set_prop_cycle('color', clist)
 
+    @cbook.deprecated("2.0")
     def ishold(self):
-        """return the HOLD status of the axes"""
+        """return the HOLD status of the axes
+
+        The `hold` mechanism is deprecated and will be removed in
+        v3.0.
```

Comment:
I would just say "in a future feature release" (as opposed to "bugfix release") in that case.